tests/test_python.py

```diff
@@ -1,13 +1,56 @@
 """Basic tests for the Python 3.6 and 3.9 base container images."""
+from pyparsing import MatchFirst
 from bci_tester.data import PYTHON310_CONTAINER
 from bci_tester.data import PYTHON36_CONTAINER
 from bci_tester.data import PYTHON39_CONTAINER
 
+import pytest
+from _pytest.mark.structures import ParameterSet
+from pytest_container import DerivedContainer
+from pytest_container.container import container_from_pytest_param
+
+wrk = "/tmp/"
+src = "tests/"
+rep = "trainers/"
+out = "output/"
+mtf = "tensorflow_examples.py"
+
+# copy tensorflow module trainer from the local repo to the container
+DOCKERF_PY_T = f"""
+WORKDIR {wrk}
+RUN mkdir {rep}
+RUN mkdir {out}
+COPY {src + rep}/{mtf}  {rep}
+"""
+
+PYTHON36_CONTAINER_T = pytest.param(
+    DerivedContainer(
+        base=container_from_pytest_param(PYTHON36_CONTAINER),
+        containerfile=DOCKERF_PY_T,
+    ),
+    marks=PYTHON36_CONTAINER.marks,
+)
+
+PYTHON39_CONTAINER_T = pytest.param(
+    DerivedContainer(
+        base=container_from_pytest_param(PYTHON39_CONTAINER),
+        containerfile=DOCKERF_PY_T,
+    ),
+    marks=PYTHON39_CONTAINER.marks,
+)
+
+PYTHON310_CONTAINER_T = pytest.param(
+    DerivedContainer(
+        base=container_from_pytest_param(PYTHON310_CONTAINER),
+        containerfile=DOCKERF_PY_T,
+    ),
+    marks=PYTHON310_CONTAINER.marks,
+)
 
 CONTAINER_IMAGES = [
-    PYTHON36_CONTAINER,
-    PYTHON39_CONTAINER,
-    PYTHON310_CONTAINER,
+    PYTHON36_CONTAINER_T,
+    PYTHON39_CONTAINER_T,
+    PYTHON310_CONTAINER_T,
```

Comment:
Please keep this one intact, the base tests should run using the unmodified images.

Reply:
The DerivedContainer was an option for the target to inject into the container some needed files for language-specific internal testing, using the COPY commands in the Dockerfile. But this implied to rebuild the base image including those files.
Should I opt for the other option to use "podman cp" command, that is at runtime, instead ?

Reply:
No, your current approach is more than fine!

Just use the `container_per_test` fixture for your tests and parametrize them individually, so that the currently existing tests will run using the unmodified containers.